server/src/main/scala/controllers/application/WithImports.scala

```diff
@@ -31,17 +34,34 @@ trait WithImports {
       log.info(s"Received a request to import red list counts")
       request.body.asJson match {
         case Some(content) =>
-          log.info(s"Received red list pax data")
+          log.info(s"Received red list pax data $content")
           Try(content.toString.parseJson.convertTo[RedListCounts])
-            .map(redListCounts => ctrl.flightsActor
-              .ask(redListCounts)
-              .map(_ => Accepted(toJson(ApiResponseBody(s"${redListCounts.counts.size} red list records imported"))))
-            ).getOrElse(Future.successful(BadRequest("Failed to parse json")))
+            .map { redListCounts =>
+              getRedListCount(redListCounts)
+                .onComplete {
+                  case Success(updatedRedListCounts) => ctrl.flightsActor
+                    .ask(RedListCounts(updatedRedListCounts))
+                  case Failure(exception) =>
+                    log.error(s"Error $exception ")
+                }
+              Future.successful(Accepted(toJson(ApiResponseBody(s"${redListCounts.passengers.size} red list records imported"))))
```

Comment:
I think we should be using the result of the above Future to determine if we give a success response or not. At the moment if we get a failure from the flights actor then we still send a 'successfully imported' response